examples/examples/rpc_middleware.rs

```diff
@@ -56,68 +61,110 @@ use jsonrpsee::ws_client::WsClientBuilder;
 pub struct CallsPerConn<S> {
 	service: S,
 	count: Arc<AtomicUsize>,
+	role: &'static str,
 }
 
 impl<'a, S> RpcServiceT<'a> for CallsPerConn<S>
 where
 	S: RpcServiceT<'a> + Send + Sync + Clone + 'static,
 {
-	type Future = BoxFuture<'a, MethodResponse>;
+	type Future = BoxFuture<'a, Result<Self::Response, Self::Error>>;
+	type Error = S::Error;
+	type Response = S::Response;
 
 	fn call(&self, req: Request<'a>) -> Self::Future {
 		let count = self.count.clone();
 		let service = self.service.clone();
+		let role = self.role;
 
 		async move {
 			let rp = service.call(req).await;
 			count.fetch_add(1, Ordering::SeqCst);
-			let count = count.load(Ordering::SeqCst);
-			println!("the server has processed calls={count} on the connection");
+			println!("{role} processed calls={} on the connection", count.load(Ordering::SeqCst));
 			rp
 		}
 		.boxed()
 	}
+
+	fn batch(&self, batch: Batch<'a>) -> Self::Future {
+		let len = batch.as_batch_entries().len();
+		self.count.fetch_add(len, Ordering::SeqCst);
+		println!("{} processed calls={} on the connection", self.role, self.count.load(Ordering::SeqCst));
+		Box::pin(self.service.batch(batch))
+	}
+
+	fn notification(&self, n: Notification<'a>) -> Self::Future {
+		Box::pin(self.service.notification(n))
+	}
 }
 
 #[derive(Clone)]
 pub struct GlobalCalls<S> {
```

Comment:
Same; a comment about what this middleware does would make the example clearer I reckon